components/net/http_loader.rs

```diff
@@ -513,14 +513,14 @@ fn request_must_be_secured(url: &Url, hsts_list: &Arc<RwLock<HSTSList>>) -> bool
 }
 
 pub fn modify_request_headers(headers: &mut Headers,
-                              doc_url: &Url,
+                              url: &Url,
                               user_agent: &str,
                               cookie_jar: &Arc<RwLock<CookieStorage>>,
                               load_data: &LoadData) {
     // Ensure that the host header is set from the original url
     let host = Host {
-        hostname: doc_url.serialize_host().unwrap(),
```

Comment:
This change doesn't respect the comment above it; we need to pass the original URL as well.